process-data/introduction.mdx

```diff
@@ -31,4 +31,20 @@ There are three types of flows:
     Flow is currently in preview, with support for one-time flows.
 </Note>
 
-To get started with Flow, see [Flows](/process-data/flows).
\ No newline at end of file
+To get started with Flow, see [Flows](/process-data/flows).
+
+## Data security
+
+When you use flows, Axiom takes the following measures to protect sensitive data such as private keys:
+
+- **Encrypted storage**: Credentials are encrypted at rest in the database. Axiom uses strong, industry-standard encryption methods and follows best practices.
+- **Per-entry encryption**: Each credential is encrypted individually with its own unique key. This limits the potential impact if any single key is compromised.
+- **Secure transit**: Credentials are encrypted in transit between your browser/client and the Axiom API using TLS 1.2 or 1.3.
+- **Internal encryption**: Credentials remain encrypted within Axiom’s internal network.
+- **Memory handling**: When credentials are briefly held in memory (for example, when delivering payloads), Axiom relies on AWS Lambda’s security guarantees and proper garbage collection.
```

Comment:
```suggestion
- **Memory handling**: When credentials are briefly held in memory (for example, when delivering payloads), Axiom relies on cloud infrastructure security guarantees and proper memory management techniques, including garbage collection.
```

I'd probably avoid calling out AWS Lambda specifically. The details of specific services may change.